src/components/table/table.module.scss

```
@@ -52,24 +140,112 @@ $HEIGHT-LARGE: 80px;
   flex-direction: column;
   gap: 4px;
   width: 100%;
+
+  &.scrollable-body {
+    flex: 1;
+    overflow-y: auto;
+    min-height: 0;
+    padding: 0 3px 3px 3px;
+    width: 100%;
+
+    .table-row {
+      min-height: $HEIGHT-DEFAULT;
+
+      &.size-small {
+        min-height: $HEIGHT-SMALL;
+      }
+
+      &.size-large {
+        min-height: $HEIGHT-LARGE;
+      }
+    }
+  }
+
+  &.horizontally-scrollable {
+    overflow-x: auto;
+    padding: 3px;
+    min-width: fit-content;
+
+    .table-row {
+      min-width: fit-content;
+
+      .table-row-content {
+        min-width: fit-content;
+        overflow: visible;
+      }
+    }
+  }
+
+  &.scrollable-body.horizontally-scrollable {
+    flex: 1;
+    overflow: visible;
+    min-height: 0;
+    padding: 0 3px 3px 3px;
+    min-width: fit-content;
+
+    .table-row {
+      min-width: fit-content;
+      min-height: $HEIGHT-DEFAULT;
+
+      &.size-small {
+        min-height: $HEIGHT-SMALL;
+      }
+
+      &.size-large {
+        min-height: $HEIGHT-LARGE;
+      }
+
+      .table-row-content {
+        min-width: fit-content;
+        overflow: visible;
+      }
+    }
+  }
```

The same. Most CSS properties can be omitted, preserving only those that are specific to the given case. 
Correct me, if I am wrong.

```suggestion
  &.horizontally-scrollable {
    overflow-x: auto;
    padding: 3px;
    min-width: fit-content;

    .table-row {
      min-width: fit-content;

      .table-row-content {
        min-width: fit-content;
        overflow: visible;
      }
    }

    &.scrollable-body {
      overflow: visible;
      padding: 0 3px 3px 3px;
    }
  }
```